diff --git a/shotstack_sdk/model_utils.py b/model_utils.py
index bb97211..eb381f6 100644
--- a/shotstack_sdk/model_utils.py
+++ b/model_utils.py
@@ -250,6 +250,7 @@ class OpenApiModel(object):
         # Get the name and value of the discriminator property.
         # The discriminator name is obtained from the discriminator meta-data
         # and the discriminator value is obtained from the input data.
+
         discr_propertyname_py = list(cls.discriminator.keys())[0]
         discr_propertyname_js = cls.attribute_map[discr_propertyname_py]
         if discr_propertyname_js in kwargs:
@@ -372,7 +373,12 @@ class OpenApiModel(object):
         # The discriminator name is obtained from the discriminator meta-data
         # and the discriminator value is obtained from the input data.
         discr_propertyname_py = list(cls.discriminator.keys())[0]
+        if not discr_propertyname_py in cls.attribute_map:
+            return None
+
         discr_propertyname_js = cls.attribute_map[discr_propertyname_py]
+
+
         if discr_propertyname_js in kwargs:
             discr_value = kwargs[discr_propertyname_js]
         elif discr_propertyname_py in kwargs:
