diff --git a/shotstack_sdk/model/render_response_data.py b/shotstack_sdk/model/render_response_data.py
index 49d7a25..8696157 100644
--- a/shotstack_sdk/model/render_response_data.py
+++ b/shotstack_sdk/model/render_response_data.py
@@ -224,7 +224,7 @@ class RenderResponseData(ModelNormal):
             if var_name not in self.attribute_map and \
                         self._configuration is not None and \
                         self._configuration.discard_unknown_keys and \
-                        self.additional_properties_type is None:
+                        self.additional_properties_type is None or var_value is None:
                 # discard variable.
                 continue
             setattr(self, var_name, var_value)
